### PR TITLE
Context Menu for Footer

### PR DESCRIPTION
**Pull request by cedricium**
This adds the context menu for #208. Currently, the menu looks like so:
screen shot 2017-09-19 at 11 06 05 am

Work that still needs to be completed: 
- [x] dynamically load menu items (see next todo) - **completed: aa65927**
- [x] use translations instead of hard-coded strings for menu items - **completed: aa65927**
- [x] `eventListener`'s for menu items:
  - [x] 'Feedback' - **completed: 4f7f72d**
  ~- [ ] 'Disable Sync'~
- [x] styling for dark theme - **completed: 8775c0b**
- [x] styling that more closely matches @ryanfeeley's design
- [x] clean up code styling (tabs -> spaces; empty lines) - **completed: 23e0989**

### CODE REVIEW

**Comment by vladikoff** [comment]
Awwww yesss!

File: scripts/postinstall.js [hunk 1]
```diff
@@ -6,7 +6,9 @@ const files = [
   copySync('node_modules/testpilot-ga/dist/index.js', 'src/vendor/testpilot-ga.js'),
   copySync('node_modules/quill/dist/quill.min.js', 'src/sidebar/vendor/quill.js'),
   copySync('node_modules/quill/dist/quill.snow.css', 'src/sidebar/vendor/quill.snow.css'),
-  copySync('node_modules/quill/LICENSE', 'src/sidebar/vendor/quill.LICENSE')
+  copySync('node_modules/quill/LICENSE', 'src/sidebar/vendor/quill.LICENSE'),
+  copySync('node_modules/material-design-lite/material.min.js', 'src/sidebar/vendor/material.js'),
+  copySync('node_modules/material-design-lite/material.min.css', 'src/sidebar/vendor/material.css')
```
Comment (pdehaan) [inline scripts/postinstall.js]: Hhmm, do we need to copy the **material-design-lite**'s LICENSE file here as well?

File: src/sidebar/panel.html [hunk 2]
```diff
@@ -5,6 +5,9 @@
   <title>Firefox Notes</title>
   <link href="vendor/quill.snow.css" rel="stylesheet">
   <link href="styles.css" rel="stylesheet">
+
+	<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
```
Comment (pdehaan) [inline src/sidebar/panel.html]: Indenting on these 2 new lines looks a bit off...

Also, possible to bundle the font family into the add-on? It seems unfortunate to have to proxy everything from the 3rd party fonts.googleapis.com site.

/cc @vladikoff

File: src/sidebar/panel.html [hunk 3]
```diff
@@ -42,13 +45,24 @@
 
     <div id="footer-buttons">
       <button id="enable-sync"></button>
-      <a target="_blank" id="give-feedback"></a>
+			<div class="wrapper">
+				<button id="demo-menu-top-right"
+        class="mdl-js-button">
+					<i class="material-icons">more_vert</i>
```
Comment (pdehaan) [inline src/sidebar/panel.html]: Sorry, I'm not super familiar w/ Material, but what is "more_vert" here? I imagine it'd possibly cause some issues for screen readers, unless the CSS is tweaking a bunch of ARIA settings or something.

File: src/sidebar/styles.css [hunk 4]
```diff
@@ -109,6 +110,7 @@ footer {
   display: flex;
   flex: 0 0 40px;
 }
+*/
```
Comment (pdehaan) [inline src/sidebar/styles.css]: Are we not using this selector anymore? Should we just ✂️ it?

File: src/sidebar/styles.css [hunk 5]
```diff
@@ -177,3 +179,7 @@ footer {
 .forbid-cursor {
   cursor: no-drop;
 }
+
+.mdl-js-button {
+  height: 100%;
+}
```
Comment (pdehaan) [inline src/sidebar/styles.css]: Again, sorry,  not super familiar w/ Material Design, but this feels pretty global and specific and could cause some future 🔥.

Should we add a more generic class name here as well so we can style _just_ that one element instead of all buttons globally (even though we only use one at the moment)?

```jsx
<button id="demo-menu-top-right" class="mdl-js-button more-button">
  <i class="material-icons">more_vert</i>
</button>
```

File: src/sidebar/panel.html [hunk 6]
```diff
@@ -42,13 +45,24 @@
 
     <div id="footer-buttons">
       <button id="enable-sync"></button>
-      <a target="_blank" id="give-feedback"></a>
+			<div class="wrapper">
+				<button id="demo-menu-top-right"
```
Comment (pdehaan) [inline src/sidebar/panel.html]: Indenting here looks a bit funny too.

Also, should the button's ID be a bit more... informative (or removed if we're not using it)?

**Comment by pdehaan** [comment]
As always, awesome work @cedricium! Thanks for all the help!

File: src/sidebar/panel.js [hunk 7]
```diff
@@ -156,14 +156,23 @@ quill.on('text-change', () => {
...
 const closeButton = document.getElementById('close-button');
 enableSync.textContent = browser.i18n.getMessage('syncNotes');
 syncNoteBody.textContent = browser.i18n.getMessage('syncNotReady2');
-giveFeedback.setAttribute('title', browser.i18n.getMessage('giveFeedback'));
-giveFeedback.setAttribute('href', SURVEY_PATH);
+
+const giveFeedback = document.getElementById('give-feedback');
+giveFeedback.textContent = browser.i18n.getMessage('feedback');
+giveFeedback.addEventListener('click', () => {
+  browser.tabs.create({
```
Comment (vladikoff) [inline src/sidebar/panel.js]: Can we avoid adding "tabs" permission?

Comment (cedricium) [reply src/sidebar/panel.js]: For as simple as it seems, I couldn't get the `giveFeedback` element (`<li>`) to open a **new tab** in the **same window**. I was able to do the following, which none seem correct for opening the survey in terms of UX:

- same window, current tab (user's current tab gets overridden)
- new window w/ one tab (happened when `<a target="_blank">` was nested inside `<li>` element) (breaks user's workflow w/ having a new window open)

I may have missed something though so please let me know if you know of a better way.

**Comment by cedricium** [comment]
@pdehaan thanks for the quick review! I started off with my IDE setup to use tabs so that's the reason for the weird spacing. I'll definitely add material's (MDL) LICENSE. As for selectors, I agree, will change over to targeting specific elements with id's as opposed to the global element.

And for `more_vert`, that's a specific icon (the 3 stacked dots). Since we're only using that, I could download that resource and drop the `<link href="https://fonts.googleapis.com/icon?family=Material+Icons">` altogether.

**Comment by cedricium** [comment]
After the latest commit, this is what the menu looks like:
context-menu

File: src/sidebar/panel.js (hunk 7 above)
Comment (vladikoff) [reply src/sidebar/panel.js]: >  I couldn't get the giveFeedback element (<li>) to open a new tab in the same window.

Don't worry about changing this. the way the current feedback window opens is fine. We cannot introduce a new permission just for this.

File: package.json [hunk 8]
```diff
@@ -8,6 +8,7 @@
     "url": "https://github.com/mozilla/notes/issues"
   },
   "dependencies": {
+    "material-design-lite": "^1.3.0",
```
Comment (vladikoff) [inline package.json]: Let's lock this to just `1.3.0`

File: src/manifest.json [hunk 9]
```diff
@@ -22,6 +22,7 @@
   },
   "permissions": [
     "storage",
+    "tabs",
```
Comment (vladikoff) [inline src/manifest.json]: let's remove this

File: src/sidebar/panel.html [hunk 10]
```diff
@@ -4,6 +4,7 @@
   <meta charset="utf-8" />
   <title>Firefox Notes</title>
   <link href="vendor/quill.snow.css" rel="stylesheet">
+  <link rel="stylesheet" href="vendor/material.css">
```
Comment (vladikoff) [inline src/sidebar/panel.html]: to keep the same code style as below put "href" first then "rel".  Also can we put it BEFORE the quill CSS so hopefully nothing in `material` overrides `quill`

File: src/sidebar/panel.html [hunk 11]
```diff
@@ -42,13 +43,20 @@
 
     <div id="footer-buttons">
       <button id="enable-sync"></button>
-      <a target="_blank" id="give-feedback"></a>
+      <div class="wrapper">
+        <button id="context-menu-button" class="mdl-js-button"></button>
+        <ul class="mdl-menu mdl-menu--top-right mdl-js-menu context-menu" data-mdl-for="context-menu-button">
+          <li id="give-feedback" class="mdl-menu__item context-menu-item"></li>
+          <li id="disable-sync" class="mdl-menu__item context-menu-item"></li>
```
Comment (vladikoff) [inline src/sidebar/panel.html]: Let's remove this button and just have 1 "Feedback" button. the "sync" button will be added as part of a different PR.

File: src/sidebar/styles.css (hunk 4 above)
Comment (vladikoff) [reply src/sidebar/styles.css]: 👍

File: src/sidebar/styles.css [hunk 12]
```diff
@@ -177,3 +179,25 @@ footer {
...
+  font-family: inherit;
+  padding: 0;
+}
+
+.context-menu-item {
+  font-size: 12px;
+  padding: 0 0 0 15px;
+  height: 18px;
+  line-height: 30px;
+}
```
Comment (vladikoff) [inline src/sidebar/styles.css]: need a new line at the of the file

**Comment by cedricium** [comment]
@vladikoff should I change the color of the Feedback button for the dark theme or leave it as is?
screen shot 2017-09-20 at 8 50 58 am

**Comment by ryanfeeley** [comment]
@cedricium It looks like [Pocket switches menus](https://sriramramani.wordpress.com/2012/12/06/runtime-theme-change/) on Android dark theme. I suppose we should too? Do you have the colors you need?

**Comment by cedricium** [comment]
@ryanfeeley should I use the dark theme colors that Notes is using or use the colors outlined in that article?

**Comment by vladikoff** [comment]
> use the dark theme colors that Notes is using 

is correct 👍  :)

**Comment by cedricium** [comment]
Having one issue with completing the dark theme for the menu. When the menu is opened, the menu item flashes it's original whitish color before the dark background is applied:
gif

@vladikoff any ideas on how to fix this? Been toying around all day and I can't find a solution.

**Comment by vladikoff** [comment]
@cedricium is it the background that flashes, or is it the effect of opening the menu? maybe you have to modify more CSS, the amation CSS. I will check it out in a bit.

**Comment by cedricium** [comment]
@vladikoff it appears to me like it's the `.mdl-menu__container` element. Try inspecting [this page](https://getmdl.io/components/index.html#menus-section) and playing around with the DOM, you'll see what I mean. I think you're right that it involves an the animation.

**Comment by cedricium** [comment]
@vladikoff I think I got it! How's this?
gif

**Comment by vladikoff** [comment]
@cedricium looks good! 👍

**Comment by cedricium** [comment]
I'm going to go ahead and say this PR is no longer a WIP and good to go after a final review.

**Comment by vladikoff** [comment]
Thanks @cedricium !! 👍
